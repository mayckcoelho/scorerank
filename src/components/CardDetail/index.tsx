import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { useSelector } from 'react-redux'
import Card from "../Card";
import Icon from "react-native-vector-icons/FontAwesome";
Icon.loadFont();

import {
    CardTitleView,
    CardTitle,
    Separator,
    CardDescription,
    BottomText,
    CardView,
    SubText,
    Description,
    ImageContent
} from './styles';

interface Props {
    icon: string;
    color: string;
    background?: string;
    title: string;
    description: string;
    image: string;
    sub: string;
    subLineThrough?: boolean;
    imageDescription: string;
    bottom: string;
    onPress(): void;
}

const CardDetail: React.FC<Props> = ({ icon, color, background, title, description, image, sub, subLineThrough, imageDescription, bottom, onPress }) => {

    return (
        <Card background={background}>
            <CardTitleView>
                <Icon name={icon} color={color} size={22} />
                <CardTitle style={{ color: color }}>{title}</CardTitle>
            </CardTitleView>
            <Separator />
            <CardDescription>{description}</CardDescription>
            <CardView>
                <ImageContent source={{ uri: image }} resizeMode="contain" />
                <View>
                    <SubText style={{ textDecorationLine: subLineThrough ? "line-through" : "none" }}>{sub}</SubText>
                    <Description>{imageDescription}</Description>
                </View>
            </CardView>
            <TouchableOpacity onPress={onPress}>
                <BottomText>{bottom}</BottomText>
            </TouchableOpacity>
        </Card>
    );
}

export default CardDetail;