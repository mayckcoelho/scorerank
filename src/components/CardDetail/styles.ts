import styled from "styled-components/native";

export const CardTitleView = styled.View`
    flex-direction: row;
    align-items: center;
`

export const CardTitle = styled.Text`
    font-size: 18px;
    margin-left: 10px;
`

export const Separator = styled.View`
    background-color: #676767;
    width: 100%;
    height: 1px;
    margin: 20px 0;
`

export const CardDescription = styled.Text`
    font-size: 18px;
    color: #676767;
`

export const BottomText = styled.Text`
    font-size: 18px;
    color: #6a1e55;
    align-self: center;
    font-weight: bold;
    margin-right: 5px;
`

export const CardView = styled.View`
    margin: 30px 0;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
`

export const SubText = styled.Text`
    font-size: 16px;
    /* text-decoration: line-through; */
    align-self: flex-end;
    color: #b2b2b2;
`

export const Description = styled.Text`
    margin: 5px 0;
    font-size: 20px;
    align-self: flex-end;
    color: #757575;
`

export const ImageContent = styled.Image`
    width: 120px;
    height: 70px;
`