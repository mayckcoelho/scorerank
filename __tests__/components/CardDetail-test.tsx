import React from 'react';
import CardDetail from '../../src/components/CardDetail';
import renderer from 'react-test-renderer';

describe('CardDetail should render correctly', () => {
    it('match snapshot', () => {
        const cardDetail = renderer
            .create(<CardDetail
                icon="close"
                color="#fff"
                title="Test"
                description="Test"
                image=""
                sub=""
                imageDescription="Test"
                bottom="Test"
                onPress={() => { }} />)
            .toJSON();
        expect(cardDetail).toMatchSnapshot();
    });
})