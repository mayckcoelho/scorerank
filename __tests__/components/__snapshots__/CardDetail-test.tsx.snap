// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetail should render correctly match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 6,
        "marginBottom": 20,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 20,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "shadowColor": "rgba(0, 0, 0, .3)",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
        "width": "95%",
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ÔÄç
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
            "marginLeft": 10,
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#676767",
          "height": 1,
          "marginBottom": 20,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 20,
          "width": "100%",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#676767",
          "fontSize": 18,
        },
      ]
    }
  >
    Test
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 30,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 30,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Array [
          Object {
            "height": 70,
            "width": 120,
          },
        ]
      }
    />
    <View>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "color": "#b2b2b2",
              "fontSize": 16,
            },
            Object {
              "textDecorationLine": "none",
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "color": "#757575",
              "fontSize": 20,
              "marginBottom": 5,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 5,
            },
          ]
        }
      >
        Test
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#6a1e55",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginRight": 5,
          },
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;
